packages/dx-react-common/docs/guides/performance-optimization.md

```diff
@@ -0,0 +1,29 @@
+# React Components - Performance Optimization
+
+This article describes the main rules you should follow to provide high perfromance for your application. For more information on performance optimization, refer to the [React guidelines](https://reactjs.org/docs/optimizing-performance.html).
+
+## Use Immutable Data Structures
+
+React Controls do not change data passed through a state and use memoization and [React.PureComponents](https://reactjs.org/docs/react-api.html#reactpurecomponent). A `React.PureComponent` shallowly compares old and new state values. In this case, if you update a state object field, React does not update the component because it compares two references to the same object. Immutable data structures help to solve this problem.
+
+Keep the Control’s state management plugins as separate components and update their properties independently to optimize performance. This allows the Control to avoid rendering unchanged UI elements.
+
+In this case, you can use a state management library like [Redux](https://redux.js.org/) to work with independent components’ states.
+
+The following demo shows how the [React Grid](https://devexpress.github.io/devextreme-reactive/react/grid) works with immutable data using Redux, [react-redux](https://github.com/reduxjs/react-redux) and [seamless-immutable](https://github.com/rtfeldman/seamless-immutable) libraries:
+
+.embedded-demo({ "path": "grid-immutability/seamless-immutable", "showThemeSelector": true })
+
+## Avoid Declaring Statements Inside Render Methods
+
+Do not declare functions and variables inside render methods. Otherwise, these functions and variables are declared each time a component is updated. One of the most common mistakes is when a developer declares a custom template within a render method. The following example, demonstrates how to correctly override a default appointment template in the [React Scheduler](https://devexpress.github.io/devextreme-reactive/react/scheduler) control:
```

Comment:
One of the most common mistakes is when a developer declares a custom template within a render method - AWK

Reply:
how to correctly override -> how to override

Reply:
The following example, demonstrates - Why is there a comma?